### DISCUSSION TITLE
2008/05/27/%E5%85%A8%E5%B1%80%E5%8F%98%E9%87%8F

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#220] [Announcements]
# 2008/05/27/%E5%85%A8%E5%B1%80%E5%8F%98%E9%87%8F

全局变量是一种数据在不同节点，不同VI，不同线程间传递的方式。数据被保存在某一固定的内存空间里，不随数据线流动。在需要读写数据的地方，不需要外部链接的数据线，直接通过某些节点或VI就可以得到目标数据，并对其操作。

https://blog.qizhen.xyz/2008/05/27/%E5%85%A8%E5%B1%80%E5%8F%98%E9%87%8F

**Comment by ruanqizhen** [comment]
之前博客下的留言：

abby
2009年08月12日 19:56 编辑
今天用labview自带的全局变量和自己写的function gloabl做了俩个程序，验证了下运行效率和内存消耗，似乎后者的消耗是前者的几百倍，大出乎我的意料，不知道是不是我的function gloabl有问题，想让您帮着看一下。如有时间，请回复我haofang021@126.com.我将程序发给您，谢谢！

回复
Qizhen
2009年01月8日 21:20 编辑
function global 中的数据在停止VI运行时不会丢失，但关闭VI后就会失去。function global可以用在多个VI中。

回复
gufei
2009年01月7日 20:16 编辑
看来好像是哦：0

回复
gufei
2009年01月7日 20:15 编辑
即function global是不是只能在一个vi中使用？

回复
gufei
2009年01月7日 19:48 编辑
请问，功能性全局变量的生存周期是不是只能是本vi啊？本vi一停止，那这块数据内存就释放了啊。。。可否指教一下？如果我做菜单形式的程序，点击一个ITEM，动态调用一个子VI读入数据（使用功能全局变量），那这个子vi一关，那不就读不到这个数据了？在线等答案，先谢谢了啊：）

回复
Qizhen
2009年01月7日 10:17 编辑
我写的比较慢，等写完了会通知大家的。估计至少今年下半年才可能看到书吧。还没起名。

回复
gufei
2009年01月6日 09:35 编辑
请问高人，你的书什么时候才能买到啊？叫啥名？望眼欲穿啊

回复
lijuansophie
2008年07月17日 16:33 编辑
受教了，实在是高人！！

回复
静
2008年06月3日 13:21 编辑
很受用，谢谢！

回复
磊
2008年05月27日 14:45 编辑
先顶了再慢慢看，长知识

回复
